server/routerlicious/packages/services/src/mongoExceptionRetryRules/mongoError/index.ts

```diff
@@ -298,6 +311,42 @@ class ConnectionClosedMongoErrorRule extends BaseMongoExceptionRetryRule {
 	}
 }
 
+class ConnectionTimedOutBulkWriteErrorRule extends BaseMongoExceptionRetryRule {
+	private static readonly errorName = "MongoBulkWriteError";
+	protected defaultRetryDecision: boolean = true;
+
+	constructor(retryRuleOverride: Map<string, boolean>) {
+		super("ConnectionTimedOutBulkWriteErrorRule", retryRuleOverride);
+	}
+
+	public match(error: any): boolean {
+		return (
+			error.name &&
+			(error.name as string) === ConnectionTimedOutBulkWriteErrorRule.errorName &&
+			error.message &&
+			/^connection .*timed out$/.test(error.message as string) === true
+		);
+	}
+}
+
+class MongoServerSelectionErrorRule extends BaseMongoExceptionRetryRule {
+	private static readonly errorName = "MongoServerSelectionError";
```

Comment:
Same what is this error and why we need to retry this error?

Reply:
Example error : `{"reason"=>{"type"=>"ReplicaSetNoPrimary", "servers"=>{}, "stale"=>false, "compatible"=>true, "heartbeatFrequencyMS"=>10000, "localThresholdMS"=>15, "setName"=>"globaldb", "maxElectionId"=>nil, "maxSetVersion"=>nil, "commonWireVersion"=>0, "logicalSessionTimeoutMinutes"=>nil}, "name"=>"MongoServerSelectionError", "message"=>"connection <monitor> to 20.205.82.1:10255 closed", ...`

This looks like a connection error from the message "connection closed", and therefore I added the retry. Again, I am open to hear your thoughts if you think otherwise. :)